src/main/java/org/jabref/logic/shared/DBMSConnection.java

```diff
@@ -21,23 +22,30 @@
     private final Connection connection;
     private final DBMSConnectionProperties properties;
 
+    public DBMSConnection(DBMSConnectionProperties connectionProperties) throws SQLException, InvalidDBMSConnectionPropertiesException {
 
-    public DBMSConnection(DBMSConnectionProperties properties) throws SQLException, InvalidDBMSConnectionPropertiesException {
-
-        if (!properties.isValid()) {
+        if (!connectionProperties.isValid()) {
             throw new InvalidDBMSConnectionPropertiesException();
         }
-        this.properties = properties;
+        this.properties = connectionProperties;
 
         try {
             DriverManager.setLoginTimeout(3);
             // ensure that all SQL drivers are loaded - source: http://stackoverflow.com/a/22384826/873282
             // we use the side effect of getAvailableDBMSTypes() - it loads all available drivers
             DBMSConnection.getAvailableDBMSTypes();
 
-            this.connection = DriverManager.getConnection(
-                    properties.getType().getUrl(properties.getHost(), properties.getPort(), properties.getDatabase()),
-                    properties.getUser(), properties.getPassword());
+            Properties props = new Properties();
```

Comment:
Can you please move these additions to the `DBMSConnectionProperties ` class as two methods `Properties  asJavaProperties()` and `String getUrl`.